repose-aggregator/components/filters/cors-filter/src/main/scala/org/openrepose/filters/cors/CorsFilter.scala

```diff
@@ -166,9 +159,42 @@ class CorsFilter @Inject()(configurationService: ConfigurationService)
 
   override def isInitialized: Boolean = initialized
 
+  def determineRequestType(request: HttpServletRequestWrapper): RequestType = {
+    val originHeader = request.getHeader(CorsHttpHeader.ORIGIN)
+    val isOptionsRequest = request.getMethod == HttpMethod.OPTIONS
+    val preflightRequestedMethod = request.getHeader(CorsHttpHeader.ACCESS_CONTROL_REQUEST_METHOD)
+    lazy val originUri = getOriginUri(originHeader)
+    lazy val hostUri = getHostUri(request)
+
+    (Option(originHeader), isOptionsRequest, Option(preflightRequestedMethod)) match {
+      case (None, isOptions, _) =>
+        logger.trace("Request is Non-CORS request because it does not have an Origin header.")
+        NonCorsRequest(isOptions)
+      case (Some(origin), true, Some(requestedMethod)) =>
+        logger.debug("Request is CORS Preflight request because it is an OPTIONS request with Origin and preflight headers.")
+        PreflightCorsRequest(origin, requestedMethod)
+      case (Some(_), isOptions, _) if originUri.isFailure =>
```

Comment:
I wonder if it would be worth inlining the `getOriginUri` function call here. `lazy val`s do have a small cost since they have to be thread safe, though I think the Scala compiler folks have made it pretty negligible at this point.